gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/jupiter/handlers/api/SyncApiReactor.java

```diff
@@ -122,6 +127,8 @@ public SyncApiReactor(
         this.resourceLifecycleManager = resourceLifecycleManager;
         this.policyManager = policyManager;
         this.groupLifecycleManager = groupLifecycleManager;
+        this.httpRequestTimeout = httpRequestTimeout;
+        this.httpRequestTimeoutGraceDelay = configuration.getProperty("http.requestTimeoutGraceDelay", Long.class, 30L);
```

Comment:
I would suggest to give both timeout on the constructor, to centralize how you retrieve them from configuration. You could for example instead of declaring a bean int, create an internal object "HttpRequestTimeout" with both value in it.

Reply:
done in https://github.com/gravitee-io/gravitee-api-management/pull/2224/commits/10c718e2cb8c3dcfb498664de8c2d42f9e3271f7